File: app/src/main/java/org/mifos/mobilebanking/utils/CurrencyUtil.java
package org.mifos.mobilebanking.utils;

import android.content.Context;
import android.telephony.TelephonyManager;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.Locale;

/**
 * Created by dilpreet on 21/8/17.
 */

public class CurrencyUtil {

    private static Locale defaultLocale = Locale.US;


    public static String formatCurrency(Context context, String amt) {
        return getDecimalFormatter(context).format(amt);
    }

    public static String formatCurrency(Context context, long amt) {
        return getDecimalFormatter(context).format(amt);
    }

    public static String formatCurrency(Context context, double amt) {
        return getDecimalFormatter(context).format(amt);
    }


    private static DecimalFormat getDecimalFormatter(Context context) {
        DecimalFormat currencyFormatter;
        Locale locale;
        TelephonyManager tm = (TelephonyManager) context.
                getSystemService(Context.TELEPHONY_SERVICE);
        try {
            locale = new Locale("en", tm.getNetworkCountryIso().toUpperCase());
        } catch (Exception e) {
            locale = defaultLocale;
        }

        try {
            currencyFormatter = (DecimalFormat) DecimalFormat.getCurrencyInstance(locale);
        } catch (Exception e) {
            currencyFormatter = (DecimalFormat) DecimalFormat.getCurrencyInstance(defaultLocale);
        }
        DecimalFormatSymbols decimalFormatSymbols = currencyFormatter.getDecimalFormatSymbols();
        decimalFormatSymbols.setCurrencySymbol("");
        currencyFormatter.setDecimalFormatSymbols(decimalFormatSymbols);
        return currencyFormatter;
    }
}

File: app/src/main/java/org/mifos/mobile/utils/CurrencyUtil.kt
package org.mifos.mobile.utils

import android.content.Context
import android.telephony.TelephonyManager
import java.text.DecimalFormat
import java.util.Locale

/**
 * Created by dilpreet on 21/8/17.
 */
object CurrencyUtil {

    private val defaultLocale = Locale.US
    fun formatCurrency(context: Context, amt: String?): String {
        return getDecimalFormatter(context).format(amt ?: "0.0")
    }

    fun formatCurrency(context: Context, amt: Long): String {
        return getDecimalFormatter(context).format(amt)
    }

    @JvmStatic
    fun formatCurrency(context: Context?, amt: Double?): String {
        return getDecimalFormatter(context).format(amt ?: 0.0)
    }

    private fun getDecimalFormatter(context: Context?): DecimalFormat {
        val currencyFormatter: DecimalFormat
        val locale: Locale?
        val tm = context?.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
        locale = try {
            Locale("en", tm.networkCountryIso.uppercase(Locale.ROOT))
        } catch (e: Exception) {
            defaultLocale
        }
        currencyFormatter = try {
            DecimalFormat.getCurrencyInstance(locale) as DecimalFormat
        } catch (e: Exception) {
            DecimalFormat.getCurrencyInstance(defaultLocale) as DecimalFormat
        }
        val decimalFormatSymbols = currencyFormatter.decimalFormatSymbols
        decimalFormatSymbols.currencySymbol = ""
        currencyFormatter.decimalFormatSymbols = decimalFormatSymbols
        return currencyFormatter
    }

    fun formatCurrency(amt: Double?): String {
        return getDecimalFormatter().format(amt ?: 0.0)
    }

    private fun getDecimalFormatter(): DecimalFormat {
        val currencyFormatter: DecimalFormat = try {
            DecimalFormat.getCurrencyInstance(defaultLocale) as DecimalFormat
        } catch (e: Exception) {
            DecimalFormat.getCurrencyInstance(Locale.getDefault()) as DecimalFormat
        }
        val decimalFormatSymbols = currencyFormatter.decimalFormatSymbols
        decimalFormatSymbols.currencySymbol = ""
        currencyFormatter.decimalFormatSymbols = decimalFormatSymbols
        return currencyFormatter
    }
}
